parser/tree-sitter-darklang/grammar.js

```diff
@@ -0,0 +1,75 @@
+module.exports = grammar({
+  name: 'darklang',
+
+  // Annoyingly, it seems to be best to write these 'top-most first'
+  rules: {
+    source_file: $ => repeat($.fn_def),
+
+    // Function definitions
+    fn_def: $ => seq(
+      'let',
+      field('name', $.identifier),
+      field('params', $.fn_params_def),
+      ':',
+      field('return_type', $.type),
+      '=',
+      field('body', $.expression)
+    ),
+
+    // fns need at least one param
+    // TODO: should this rule be enforced like this _in the parser_?
+    // TODO: we could just repeat1 here
+    fn_params_def: $ => seq(
+      field("first", $.fn_param_def),
+      field("additional", repeat($.fn_param_def))
+    ),
```

Comment:
I'm probably over-complicating things by trying to enforce such a rule in the grammar? Regardless, this particular set-up makes the SyntaxTree->PT.UserFunction process awkward.